### DISCUSSION TITLE
integration with  react-hook-form, how to validate ?

### DISCUSSION THREAD

**Discussion by zoutigo** [react-page/react-page#1095] [Q&A]
Hello.

Everything is working fine except validation.
I would like to know how to validate limit of typed characters , or how to check that the Editor have been filled.

i don't know what to check, in my value , since the editor in based on an object.

```
  <Controller
            name="content"
            control={control}
            defaultValue={page ? JSON.parse(page.content) : null}
            rules={{
              required: 'il veillez ajouter un texte',
              validate:(value)=> ??????
            }}
            render={({ field }) => (
              <Editor
                {...field}
                cellPlugins={cellPlugins}
                cellSpacing={cellSpacing}
                lang="fr"
                onChange={(newvalue) => field.onChange(newvalue)}
              />
            )}
          />
```

**Comment by macrozone** [comment]
hi @zoutigo , that is a good question.

Its ambiguous what an app would consider an "empty" editor. Is it enough to contain a cell with any plugin (e.g. an image) ? does it need to have at least one slate (richtext) cell? this is app dependent.

The `value` that you get has a type `Value` that should make it a bit easier to understand how it is structured.

having said that, there is a function that can walk the whole value, but its not documented yet. That might make it easier to e.g. check whether a certain cell exists in the tree

**Reply by macrozone** [reply]
hi @zoutigo, i added a helper function which makes it easier to get the plaintext content from the editor:


https://react-page.github.io/docs/#/utils?id=gettextcontents-get-raw-text-contents

**Reply by zoutigo** [reply]
hello.
Great, i assume it means uninstalling the previous version and re-installing . 'cause i'm currently getting ****'getTextContents' is not exported from '@react-page/editor'****
I'm going to try it and give you the feedback.

**Reply by zoutigo** [reply]
i've re-installed, but still getting the same error about the export from  @react-page/editor

**Reply by macrozone** [reply]
@zoutigo which version did you install? (check package.json and yarn.lock or package-lock

**Reply by zoutigo** [reply]
It's 4.2.0

**Reply by macrozone** [reply]
can you show a reproduction? just checked and it works.
